src/core/dir.c

```diff
@@ -68,19 +64,45 @@ const dir_listing *dir_find_files_with_extension(const char *dir, const char *ex
 {
     clear_dir_listing();
     platform_file_manager_list_directory_contents(dir, TYPE_FILE, extension, add_to_listing);
-    qsort(data.listing.files, data.listing.num_files, sizeof(char *), compare_lower);
+    dir_sort_by_filename();
     return &data.listing;
 }
 
 const dir_listing *dir_find_all_subdirectories(void)
 {
     clear_dir_listing();
     platform_file_manager_list_directory_contents(0, TYPE_DIR, 0, add_to_listing);
-    qsort(data.listing.files, data.listing.num_files, sizeof(char *), compare_lower);
+    dir_sort_by_filename();
     return &data.listing;
 }
 
-static int compare_case(const char *filename)
+static int compare_lower(const void *va, const void *vb)
+{
+    const dir_entry* a = (const dir_entry*) va;
+    const dir_entry* b = (const dir_entry*) vb;
+
+    return platform_file_manager_compare_filename(a->name, b->name);
+}
+
+void dir_sort_by_filename(void)
+{
+    qsort(data.listing.files, data.listing.num_files, sizeof(data.listing.files[0]), compare_lower);
+}
+
+static int compare_modified_time(const void *va, const void *vb)
+{
+    const dir_entry* a = (const dir_entry*) va;
+    const dir_entry* b = (const dir_entry*) vb;
+
+    return a->modified_time < b->modified_time;
```

Comment:
To fix sorting based on modified time, please change this:

```suggestion
        if (a->modified_time < b->modified_time) {
        return 1;
    } else if (a->modified_time == b->modified_time) {
        return 0;
    } else {
        return -1;
    }
```

Reply:
Oops. That's a biggie.